File: API/Core/DTOs/Recommendations/RecommendationDTO.cs
namespace Core.DTOs.Recommendations
{
    public class RecommendationDTO
    {
        public int ProductId { get; set; }
        public string ProductName { get; set; }
        public string ProductSlug { get; set; }
        public decimal MinPrice { get; set; }
        public decimal MaxPrice { get; set; }
        public string PriceDisplay { get; set; }
        public string ImageUrl { get; set; }
        public int TotalStock { get; set; }
        public List<string> AvailableColors { get; set; }
        public List<string> AvailableSizes { get; set; }
        public double Score { get; set; }
        public string ReasonCode { get; set; }
        public string ReasonText { get; set; }
    }
}

File: ClientUI/src/app/_shared/_models/recommendationDTO.ts
export interface RecommendationDto {
  productId: number;
  productName: string;
  productSlug: string;
  minPrice: number;
  maxPrice: number;
  priceDisplay: string;
  imageUrl: string;
  totalStock: number;
  availableColors: string[];
  availableSizes: string[];
  score: number;
  reasonCode: string;
  reasonText: string;
}

export interface TrackInteractionRequest {
  productId: number;
  skuId?: number;
  durationSeconds?: number;
}

export enum InteractionType {
  View = 1,
  AddToCart = 2,
  Purchase = 3
}